mock_drivers/rosbag_mock_drivers/src/USFControllerDriver.cpp

```diff
@@ -0,0 +1,249 @@
+#include "rosbag_mock_drivers/VehicleCmd.h"
+
+#include <iostream>
+#include <string>
+#include <cstdio>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <pthread.h>
+//head files required by USF controller driver
+
+#include "rosbag_mock_drivers/MockControllerDriver.h"
+
+namespace mock_drivers
+{
+std::vector<DriverType> MockControllerDriver::getDriverTypes()
+{
+  return { DriverType::CONTROLLER };
+}
+
+uint8_t MockControllerDriver::getDriverStatus()
+{
+  return cav_msgs::DriverStatus::OPERATIONAL;
+}
+
+//have a new vehicle command callback function below
+/*void MockControllerDriver::vehicleCmdCallback(const autoware_msgs::VehicleCmd::ConstPtr& msg)
+{
+  robot_enabled_ = true;  // If a command was received set the robot enabled status to true
+}*/
+
+bool MockControllerDriver::enableRoboticSrv(const cav_srvs::SetEnableRobotic::Request& req,
+                                            cav_srvs::SetEnableRobotic::Response& res)
+{
+  if (robot_enabled_ && req.set == cav_srvs::SetEnableRobotic::Request::ENABLE)
+  {
+    robot_active_ = true;
+  }
+  else
+  {
+    robot_active_ = false;
+  }
+
+  return true;
+}
+
+MockControllerDriver::MockControllerDriver(bool dummy)
+{
+  mock_driver_node_ = MockDriverNode(dummy);
+
+  robot_status_ptr_ =
+      boost::make_shared<ROSComms<cav_msgs::RobotEnabled>>(CommTypes::pub, false, 10, robot_status_topic_);
+  
+  ////vehicle command topic is subscribed by the vehiclesender node.
+
+  /*vehicle_cmd_ptr_ = boost::make_shared<ConstPtrRefROSComms<autoware_msgs::VehicleCmd>>(
+      std::bind(&MockControllerDriver::vehicleCmdCallback, this, std::placeholders::_1), CommTypes::sub, false, 10,
+      vehicle_cmd_topic_);*/
+
+  enable_robotic_ptr_ =
+      boost::make_shared<ROSComms<cav_srvs::SetEnableRobotic::Request&, cav_srvs::SetEnableRobotic::Response&>>(
+          std::bind(&MockControllerDriver::enableRoboticSrv, this, std::placeholders::_1, std::placeholders::_2),
+          CommTypes::srv, enable_robotic_srv_);
+}
+
+bool MockControllerDriver::onSpin()
+{
+  cav_msgs::RobotEnabled robot_status;
+  robot_status.robot_active = robot_active_;
+  robot_status.robot_enabled = robot_enabled_;
+
+  mock_driver_node_.publishDataNoHeader<cav_msgs::RobotEnabled>(robot_status_topic_, robot_status);
+
+  return true;
+}
+
+unsigned int MockControllerDriver::getRate()
+{
+  return 20;  // 20 Hz as default spin rate to match expected controller data rate
+}
+
+//USF controller message structure
+struct CommandData {
+  double linear_x;
+  double angular_z;
+
+  void reset();
+};
+
+void CommandData::reset()
+{
+  linear_x      = 0;
+  angular_z     = 0;
+}
+
+static CommandData command_data;
+
+static void vehicleCmdCallback(const autoware_msgs::VehicleCmd& msg)
+{
+  command_data.linear_x = msg.twist_cmd.twist.linear.x;
+  command_data.angular_z = msg.twist_cmd.twist.angular.z;
+}
+
+static void *sendCommand(void *arg)
```

Comment:
Is there a reason this accepts a `void *`, then forces a cast immediately to an `int*`, then dereferences that into an int? Would it not be simpler to simply accept the int directly as a parameter?

Also, what is the purpose of the return type here? It appears this function always returns nullptr, even on success? You could simply omit the need to return anything by changing the return type from `void *` to `void`. Or, if an error status should be returned you could return a bool for a simple pass/fail or an int if differentiation between the error codes matters.